src/clients/BundleDataClient/BundleDataClient.ts

```diff
@@ -926,6 +927,8 @@ export class BundleDataClient {
                     fastFillsReplacingSlowFills.push(relayDataHash);
                   }
                 }
+              } else {
+                throw new Error("Duplicate fill detected");
```

Comment:
wdym by this?

Reply:
this is just an assert. We should never see duplicate fills in memory